tests/foreman/virtwho/api/test_esx.py

```diff
@@ -82,67 +83,19 @@ def test_positive_deploy_configure_by_id(
         :CaseImportance: High
         """
         assert virtwho_config.status == 'unknown'
-        command = get_configure_command(virtwho_config.id, default_org.name)
-        hypervisor_name, guest_name = deploy_configure_by_command(
-            command, form_data['hypervisor_type'], debug=True, org=default_org.label
-        )
-        virt_who_instance = (
-            target_sat.api.VirtWhoConfig()
-            .search(query={'search': f'name={virtwho_config.name}'})[0]
-            .status
-        )
-        assert virt_who_instance == 'ok'
-        hosts = [
-            (
-                hypervisor_name,
-                f'product_id={settings.virtwho.sku.vdc_physical} and type=NORMAL',
-            ),
-            (
-                guest_name,
-                f'product_id={settings.virtwho.sku.vdc_physical} and type=STACK_DERIVED',
-            ),
-        ]
-        for hostname, sku in hosts:
-            host = target_sat.cli.Host.list({'search': hostname})[0]
-            subscriptions = target_sat.cli.Subscription.list(
-                {'organization': default_org.name, 'search': sku}
+        if deploy_type == "id":
```

Comment:
The changes look good some suggestions you can try.

Nonblocking: I see you updated the conditional statement for all the test cases maybe you could write a helper method that will be invoked according to the deploy_type argument in a helper_file where you have kept all the fixtures.

Reply:
@adarshdubey-star Thanks for your good suggestions, In fact only the deploy type by id and by script cases will use the deploy_type, other cases will not. but it is good method to reuse the helper, if there are some demo for this update? Thank you very much.